src/components/Button/Button.tsx

```diff
@@ -1,41 +1,40 @@
-// @flow
-import type { UIState } from '../../constants/types'
-import React, { PureComponent as Component } from 'react'
+import * as React from 'react'
 import getValidProps from '@helpscout/react-utils/dist/getValidProps'
 import { namespaceComponent } from '../../utilities/component'
 import { classNames } from '../../utilities/classNames'
 import { noop } from '../../utilities/other'
 import RouteWrapper from '../RouteWrapper'
-import { COMPONENT_KEY } from './utils'
+import { COMPONENT_KEY } from './Button.utils'
+import { UIState } from '../../constants/types'
 
-type ButtonSelector = 'a' | 'button' | 'input'
+export type ButtonSelector = 'a' | 'button' | 'input'
```

Comment:
Can you move the types to the type folder? And while you are it, rename `Props` to `ButtonProps`. Thanks!